src/workerd/api/sockets.c++

```diff
@@ -417,7 +417,7 @@ void Socket::handleProxyStatus(jsg::Lock& js, kj::Promise<kj::Maybe<kj::Exceptio
 
 void Socket::handleProxyError(jsg::Lock& js, kj::Exception e) {
   resolveFulfiller(js, kj::cp(e));
-  openedResolver.reject(js, kj::mv(e));
```

Comment:
This might be causing an internal test failure – are KJ exceptions not safe to copy/is moving and then accessing e again in L422 the right approach here?

Reply:
The change looks correct. If a test is failing it's likely because the change is exposing another possible bug or the test was written to rely on the previously broken behavior.

Reply:
That ended up being the case, thank you!